modules/charon-core/src/main/java/org/wso2/charon3/core/extensions/RoleManager.java

```diff
@@ -0,0 +1,116 @@
+import org.wso2.charon3.core.exceptions.NotImplementedException;
+import org.wso2.charon3.core.objects.Role;
+import org.wso2.charon3.core.utils.codeutils.Node;
+import org.wso2.charon3.core.utils.codeutils.SearchRequest;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * OSGi service interface which uses to manage roles.
+ */
+public interface RoleManager {
+
+    /**
+     * Create a role.
+     *
+     * @param role Role Object.
+     * @return Role.
+     * @throws CharonException         CharonException.
+     * @throws ConflictException       ConflictException.
+     * @throws NotImplementedException NotImplementedException.
+     * @throws BadRequestException     BadRequestException.
+     */
+    Role createRole(Role role) throws CharonException, ConflictException, NotImplementedException, BadRequestException;
+
+    /**
+     * Get the role for the given ID.
+     *
+     * @param id                 Role ID.
+     * @param requiredAttributes Required Attributes.
+     * @return Role.
+     * @throws NotImplementedException NotImplementedException.
+     * @throws BadRequestException     BadRequestException.
+     * @throws CharonException         CharonException.
+     * @throws NotFoundException       NotFoundException.
+     */
+    Role getRole(String id, Map<String, Boolean> requiredAttributes)
+            throws NotImplementedException, BadRequestException, CharonException, NotFoundException;
+
+    /**
+     * Delete the given role.
+     *
+     * @param id Role ID.
+     * @throws NotFoundException       NotFoundException.
+     * @throws CharonException         CharonException.
+     * @throws NotImplementedException NotImplementedException.
+     * @throws BadRequestException     BadRequestException.
+     */
+    void deleteRole(String id) throws NotFoundException, CharonException, NotImplementedException, BadRequestException;
+
+    /**
+     * List roles with Get.
+     *
+     * @param node       Node
+     * @param startIndex Start Index
+     * @param count      Count
+     * @param sortBy     Sort by
+     * @param sortOrder  Sort order
+     * @return List of roles.
+     * @throws CharonException         CharonException.
+     * @throws NotImplementedException NotImplementedException.
+     * @throws BadRequestException     BadRequestException.
+     */
+    List<Object> listRolesWithGET(Node node, Integer startIndex, Integer count, String sortBy, String sortOrder)
+            throws CharonException, NotImplementedException, BadRequestException;
+
+    /**
+     * Update the role.
+     *
+     * @param oldRole Old role.
+     * @param newRole new role.
+     * @return Updated role.
+     * @throws NotImplementedException NotImplementedException.
+     * @throws BadRequestException     BadRequestException.
+     * @throws CharonException         CharonException.
+     * @throws NotFoundException       NotFoundException.
+     */
+    Role updateRole(Role oldRole, Role newRole)
```

Comment:
shouldn't we support updateRole using role ID?

Reply:
ID is included inside the `Role` object.